connectors/services/sync.py

```diff
@@ -40,63 +45,68 @@ def __init__(self, config, args):
         super().__init__(config)
         self.args = args
         self.idling = self.service_config["idling"]
-        self.hb = self.service_config["heartbeat"]
+        self.heartbeat_interval = self.service_config["heartbeat"]
         self.concurrent_syncs = self.service_config.get(
             "max_concurrent_syncs", DEFAULT_MAX_CONCURRENT_SYNCS
         )
-        self.connectors = None
+        self.bulk_options = self.es_config.get("bulk", {})
+        self.source_klass_dict = get_source_klass_dict(config)
+        self.connector_index = None
+        self.sync_job_index = None
 
     async def _one_sync(self, connector, es, sync_now):
         if connector.native:
             logger.debug(f"Connector {connector.id} natively supported")
 
-        try:
-            await connector.prepare(self.config)
-        except ServiceTypeNotConfiguredError:
-            logger.error(
-                f"Service type is not configured for connector {self.config['connector_id']}"
-            )
-            return
-        except ConnectorUpdateError as e:
-            logger.error(e)
-            return
-        except ServiceTypeNotSupportedError:
-            logger.debug(f"Can't handle source of type {connector.service_type}")
-            return
-        except DataSourceError as e:
-            await connector.error(e)
-            logger.critical(e, exc_info=True)
-            self.raise_if_spurious(e)
+        if not await self._prepare_connector(connector):
             return
 
         try:
             # the heartbeat is always triggered
-            connector.start_heartbeat(self.hb)
+            await connector.heartbeat(self.heartbeat_interval)
```

Comment:
I'd probably leave `heartbeat_interval` out of the function and do the condition here, e.g:

```python
if self.should_trigger_heartbeat():
    connector.heartbeat()
```

Reply:
But it's more of a nit fixable later